content/integrations/postgresql.md

```diff
@@ -47,58 +47,79 @@ When it prompts for a password, enter the one used in the first command.
 
 
 ## Custom metrics
- 
-Below are some examples of commonly used metrics, which are implemented as custom metrics. Uncomment them if you want to use them as is, or use as an example for creating your own custom metrics. 
 
-The format for describing custome metrics is identical with the one used for common metrics in `postgres.py`. Be extra careful with ensuring proper custom metrics description format. If your custom metric does not work after an agent restart, look for errors in the output of `/etc/init.d/datadog-agent info` command, as well as `/var/log/datadog/collector.log` file.
+PostgreSQL custom metrics can be collected by mapping SQL query results to metrics in Datadog. This mapping is expressed using four components: descriptors, metrics, query, and relation.
+
+* **Descriptors** are a sequence of tuples matching a column name or alias from the custom metric query to a tag in Datadog. In the example below, `relname` is mapped to `schema`. Custom metrics in Datadog can then be filtered and aggregated by `schema` tag.
+* **Metrics** are mappings from a query column or column function to a tuple containing the Datadog custom metric name and metric type (`RATE`, `GAUGE`, or `MONOTONIC`). In the example below, the sum of the `idx_scan` column is mapped to the Datadog custom metric name `postgresql.idx_scan_count_by_schema`.
+* **Query** is where you'll construct the SQL query to obtain your custom metrics. Each column name in your select query should have a corresponding tuple in the Descriptors section. Each column listed in the Metrics section will be substituted for the `%s` in the query.
+* **Relation** indicates whether to include schema relations specified in the [configuration options above](#configuration-options). If set to `true`, the last `%s` in the Query will be set to the list of schema names specified in the Relations configuration option.
+
+### Example 1
 
     custom_metrics:
-    - # Londiste 3 replication lag
-      descriptors:
-        - [consumer_name, consumer_name]
-      metrics:
-        >
-          GREATEST(0, EXTRACT(EPOCH FROM lag)) as lag: 
-          [postgresql.londiste_lag, GAUGE]
-        >
-          GREATEST(0, EXTRACT(EPOCH FROM lag)) as last_seen: 
-          [postgresql.londiste_last_seen, GAUGE]
-        pending_events: [postgresql.londiste_pending_events, GAUGE]
-      query: 
-        >
-          SELECT consumer_name, %s from pgq.get_consumer_info() 
-          where consumer_name !~ 'watermark$';
-      relation: false
-    collect_function_metrics: False
-    collect_count_metrics: False
-{:.language-yaml}
+      # All index scans & reads
+      - descriptors:
+          - [relname, schema]
+        metrics:
+            SUM(idx_scan) as idx_scan_count: [postgresql.idx_scan_count_by_schema, RATE]
+            SUM(idx_tup_read) as idx_read_count: [postgresql.idx_read_count_by_schema, RATE]
+        query: SELECT relname, %s FROM pg_stat_all_indexes GROUP BY relname;
+        relation: false
```

Comment:
I think that displaying the final query made to postgres make the example much more understandable.